code/game/gamemodes/cult/cult_structures.dm

```diff
@@ -24,7 +24,27 @@
 	var/isbroken = 0
 	light_range = 5
 	light_color = LIGHT_COLOR_RED
-	var/obj/item/wepon = null
+	var/last_check
+
+/obj/structure/cult/pylon/New()
+	..()
+	processing_objects.Add(src)
+
+/obj/structure/cult/pylon/Destroy()
+	..()
+	processing_objects.Remove(src)
+
+/obj/structure/cult/pylon/process()
+	if(!isbroken && world.time > last_check + 3 SECONDS)
+		var/list/can_see = view(src, 3)
+		last_check = world.time
+		for(var/mob/living/simple_animal/construct/C in can_see)
```

Comment:
I remember range not having the optimization though?

That's why trange exists, and I even tried to make a better trange by doing a dview with full visibility settings and it performed almost identical (but still worse) than trange, and range was behind everything.